gtsfm/densify/mvs_patchmatchnet.py

```diff
@@ -0,0 +1,247 @@
+                del sample_device
+
+                ids = sample["idx"]
+
+                # save depth maps and confidence maps
+                for idx, depth_est, photometric_confidence in zip(
+                    ids, outputs["refined_depth"]["stage_0"], outputs["photometric_confidence"]
+                ):
+
+                    idx = idx.cpu().numpy().tolist()
+                    depth_est_list[idx] = depth_est.copy()
+                    confidence_est_list[idx] = photometric_confidence.copy()
+
+                logger.info(
+                    "[Densify::PatchMatchNet] Iter {}/{}, time = {:.3f}".format(
+                        batch_idx + 1, len(loader), time.time() - start_time
+                    )
+                )
+
+        # filter inference result with thresholds
+        dense_point_cloud = self.filter_depth(
+            dataset=dataset,
+            depth_list=depth_est_list,
+            confidence_list=confidence_est_list,
+            geo_pixel_thres=thresholds[0],
+            geo_depth_thres=thresholds[1],
+            photo_thres=thresholds[2],
+        )
+
+        return dense_point_cloud
+
+    def filter_depth(
+        self,
+        dataset: PatchmatchNetData,
+        depth_list: Dict[int, np.ndarray],
+        confidence_list: Dict[int, np.ndarray],
+        geo_pixel_thres: float,
+        geo_depth_thres: float,
+        photo_thres: float,
+    ) -> np.ndarray:
+        """Filter depth map and get filtered dense point cloud
+        Ref: Wang et al. https://github.com/FangjinhuaWang/PatchmatchNet/blob/main/eval.py
+
+        Args:
+            dataset (PatchmatchNetData): an instance of PatchmatchData as the inference dataset
+            depth_list (Dict[int, np.ndarray]): list of 2D depth map (H, W) from each view
+            confidence_list (Dict[int, np.ndarray]): list of 2D confidence map (H, W) from each view
+            geo_pixel_thres (float): geometric pixel threshold
+            geo_depth_thres (float): geometric depth threshold
+            photo_thres (float): photometric threshold
+
+        Returns:
+            np.ndarray: 3D coordinates (in the world frame) of the dense point cloud, (point number, 3)
+        """
+        # coordinates of the final point cloud
+        vertices = []
+        # vertex colors of the final point cloud, used in generating colored mesh
+        vertex_colors = []
+
+        pair_data = dataset.get_packed_pairs()
+
+        # for each reference view and the corresponding source views
+        for ref_view, src_views in pair_data:
+            # load the camera parameters
+            ref_intrinsics, ref_extrinsics = dataset.get_camera_params(ref_view)
+
+            # load the reference image
+            ref_img = dataset.get_image(ref_view)
+            # load the estimated depth of the reference view
+            ref_depth_est = depth_list[ref_view][0]
+            # load the photometric mask of the reference view
```

Comment:
how do we define "photometric mask"? would you mind adding that as an inline comment?